### DISCUSSION TITLE
How to use the new wry API `WebView::cookies` and `WebView::cookies_for_url`  in a tauri project?

### DISCUSSION THREAD

**Discussion by LZQCN** [tauri-apps/tauri#11655] [Q&A] [answered]
Hello

I recently discovered that the upstream library wry in version 0.47.0 has added two new APIs `WebView: WebView::cookies` and `WebView::cookies_for_url`. https://github.com/tauri-apps/wry/pull/1399#issue-2603375338

How to use these APIs in a tauri project?

**Comment by FabianLars** [comment, marked as answer]
They have to be explicitly exposed in tauri before you can use them in your apps. Similar to PRs like this https://github.com/tauri-apps/tauri/pull/11486

**Reply by LZQCN** [reply]
Oh, this indeed is a daunting task, much more complex than I had anticipated.

**Reply by murnifine** [reply]
Can this API be used to set cookies as well?

**Reply by FabianLars** [reply]
No, only getters were implemented in Wry. at least some platforms do support setting them though so this can be added in the future.
